test/services/database/user_comment_repository_test.dart

```diff
@@ -0,0 +1,174 @@
+
+    setUp(() async {
+      fakeFirestore = FakeFirebaseFirestore();
+
+      final container = ProviderContainer(
+        overrides: [
+          firestoreProvider.overrideWithValue(fakeFirestore),
+        ],
+      );
+
+      userCommentRepository =
+          container.read(userCommentRepositoryServiceProvider);
+
+      user = (await FirestoreUserGenerator.addUsers(fakeFirestore, 1)).first;
+
+      userCommentGenerator = UserCommentFirestoreGenerator();
+      userCommentDataGenerator = UserCommentDataGenerator();
+      userCommentsCollection = fakeFirestore
+          .collection(UserFirestore.collectionName)
+          .doc(user.uid.value)
+          .collection(UserCommentFirestore.userCommentSubCollectionName);
+    });
+
+    group("getting user comments", () {
+      test("should return an empty list if the user has no comments", () async {
+        final userComments =
+            await userCommentRepository.getUserComments(user.uid);
+
+        expect(userComments, isEmpty);
+      });
+
+      test("should return the user's comments", () async {
+        final userComments = await userCommentGenerator.addComments(
+          3,
+          user.uid,
+          userCommentRepository,
+        );
+
+        final fetchedUserComments =
+            await userCommentRepository.getUserComments(user.uid);
+
+        expect(fetchedUserComments, unorderedEquals(userComments));
+      });
+
+      test("should throw an error if a document has missing fields", () async {
+        final userCommentData =
+            userCommentDataGenerator.createMockUserCommentData();
+
+        final userCommentId = await userCommentRepository.addUserComment(
+          user.uid,
+          userCommentData,
+        );
+
+        // Delete the comment id field
+        await userCommentsCollection
+            .doc(userCommentId.value)
+            .update({UserCommentData.commentIdField: FieldValue.delete()});
+
+        expect(
+          () => userCommentRepository.getUserComments(user.uid),
+          throwsA(isA<Exception>()),
+        );
+      });
+    });
+
+    group("adding user comments", () {
+      test("should add a user comment", () async {
+        final userCommentData =
+            userCommentDataGenerator.createMockUserCommentData();
+
+        // Add the user comment
+        final userCommentId = await userCommentRepository.addUserComment(
+          user.uid,
+          userCommentData,
+        );
+
+        // Check if the comment was added
+        final expectedUserComment = UserCommentFirestore(
+          id: userCommentId,
+          data: userCommentData,
+        );
+
+        final fetchedUserComments =
+            await userCommentRepository.getUserComments(user.uid);
+
+        expect(fetchedUserComments, [expectedUserComment]);
+      });
+    });
+
+    group("deleting user comments", () {
+      test("should delete a single user comment", () async {
```

Comment:
Again, are you sure this test is really important? There's almost exactly the same one right after, line 138.

Reply:
You are right :+1: 
Looking back at it, I don't think this test adds anything to the robustness.

Thus I removed it in 064cd036cc03930d79298d0da7290c2494a1ef1a.